pkg/apis/nfd/v1/nodefeaturediscovery_types.go

```diff
@@ -0,0 +1,95 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodeFeatureDiscoverySpec defines the desired state of NodeFeatureDiscovery
+// +k8s:openapi-gen=true
+type NodeFeatureDiscoverySpec struct {
+	Operand OperandSpec `json:"operand"`
+	Config  ConfigSpec  `json:"nfd-worker"`
+}
+
+// OperandSpec describes configuration options for the operand
+type OperandSpec struct {
+	// +kubebuilder:validation:Pattern=[a-zA-Z0-9\.\-\/]+
+	Namespace string `json:"namespace,omitempty"`
+
+	// +kubebuilder:validation:Pattern=[a-zA-Z0-9\.\-\/]+
+	Repository string `json:"repository,omitempty"`
+
+	// +kubebuilder:validation:Pattern=[a-zA-Z0-9\-]+
+	Image string `json:"image,omitempty"`
+
+	// +kubebuilder:validation:Pattern=[a-zA-Z0-9\.-]+
+	Version string `json:"version,omitempty"`
+
+	// Image pull policy
+	// +kubebuilder:validation:Optional
+	ImagePullPolicy string `json:"imagePullPolicy,omitempty"`
+}
+
+// ConfigSpec describes configuration options for the NFD worker
+type ConfigSpec struct {
+	// BinaryData holds the NFD configuration file
+	BinaryData string `json:"config"`
```

Comment:
I'd suggest to align the naming, e.g.
```
configData string `json:"configData"`
```
or something else with the same idea...